client/dashboard/sites/settings-subscription-gifting/index.tsx

```diff
@@ -33,21 +35,20 @@ const form = {
 	fields,
 };
 
-export function getSubscriptionGiftingSettingBadges( settings: SiteSettings ) {
-	return settings.wpcom_gifting_subscription
-		? [ { text: __( 'Enabled' ), intent: 'success' as const } ]
-		: [ { text: __( 'Disabled' ) } ];
-}
-
 export default function SubscriptionGiftingSettings() {
 	const { siteSlug } = siteSettingsSubscriptionGiftingRoute.useParams();
+	const { data: features } = useQuery( siteFeaturesQuery( siteSlug ) );
 	const { data } = useQuery( siteSettingsQuery( siteSlug ) );
 	const mutation = useMutation( siteSettingsMutation( siteSlug ) );
 
-	if ( ! data ) {
+	if ( ! data || ! features ) {
 		return null;
 	}
 
+	if ( ! hasSubscriptionGiftingFeature( features ) ) {
+		return <NotFound />;
```

Comment:
Would it be better to redirect to the parent page? It seems the user cannot do anything on the not found page

Reply:
Yeah, upsell is also the best approach to me but it seems this feature might not be something you would upgrade for as mentioned above.

Reply:
Why do you need a plan to _accept_ a gift subscription?

Reply:
The gift is a payment for a plan subscription renewal of a site, which does not exist for Free sites by definition.

Reply:
It sounds like if you end up on this route/page, it's an error rather than not found.

Reply:
Which we should treat similarly to routes like `/v2/bogus`, right? Currently, we show 404 Not Found in this case 😄

Reply:
I think instead of rendering the actual not found page (which might not be correct in this context depending on layout...), it's better to actually throw `throw notFound()` (see TanStack Router docs)

Reply:
I actually had tried this approach. But we need to add a `<CatchNotFound>` wrapper somewhere, e.g. in the root component as follows:

```jsx
	return (
		<div className="dashboard-root__layout">
			{ ( isFetching > 0 || isNavigating ) && <LoadingLine /> }
			{ isInitialLoad && <LoadingLogo className="wpcom-site__logo" /> }
			<Header />
			<main>
				<CatchNotFound fallback={ NotFound }>
					<Outlet />
				</CatchNotFound>
			</main>
			<CommandPalette />
		</div>
	);
```

which would render:

<img width="755" alt="image" src="https://github.com/user-attachments/assets/3d3313d0-e507-4a4b-9e9d-e50a02ebf7b7" />

would that be a better approach?

Reply:
That would be better I think

Reply:
But I wonder why we need the fallback since we're actually defining the notFound component. Feels like tanstack should be smart enough to just use it.

Reply:
> But I wonder why we need the fallback since we're actually defining the notFound component. Feels like tanstack should be smart enough to just use it.

I wondered the same thing and I "wasted" a whole hour debugging this this morning 🤦 seems like the `defaultNotFoundComponent` is for catching 404s in routes, not errors that you `throw` (even if it's `notFound()`)

Reply:
> When might you see this 404?

A possible (but convoluted) scenario is as follows: Aa user might have a Free site and a paid site. Then, they visit the setting for the paid site. Then they manually replace the site slug with that of the free site. In this case, we wouldn't want to show any irrelevant settings.

Reply:
@youknowriad please check this: https://github.com/Automattic/wp-calypso/pull/103227/commits/d3a41a6402040e2f0feeecbd2b5941442c4b2e6c

Reply:
> Then they manually replace the site slug with that of the free site. In this case, we wouldn't want to show any irrelevant settings.

Not a strong opinion, and no blockers, but that to me is a "no settings found" use case, not a 404.

Reply:
LGTM 👍